jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java

```diff
@@ -62,10 +62,11 @@ public void setUp() throws Exception {
   public void testRemoteReporterReport() throws Exception {
     Span span = (Span) tracer.buildSpan("raza").start();
     reporter.report(span);
-    Thread.sleep(5);
+    // do sleep until automatic flush happens on 'reporter'
+    Thread.sleep(flushInterval + 20); // added 20ms on top of 'flushInterval' to avoid corner cases
```

Comment:
Another option is to use JUnit 5's [assertTimeout](https://howtoprogram.xyz/2016/12/01/junit-5-timeout-test/). However, this depends on moving the project to Java 8, and updating the JUnit dependency.